### PR TITLE
KTOR-6400 Add attribute for extending isStaticContent() functionality in Webjars

### PR DESCRIPTION
**Pull request by bjhham**
**Subsystem**
StaticContent, Webjars

**Motivation**
[KTOR-6400](https://youtrack.jetbrains.com/issue/KTOR-6400) Marking files provided by Webjars plugin as static files

**Solution**
Introduced a public attribute for any static file response producer to add the path to the call.  This allows plugins like `Webjars` (or any other) to return true for the `ApplicationCall.isStaticContent()` extension function.  Previously, this function only checked if a path parameter was present from the routing.

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/test/io/ktor/server/webjars/WebjarsTest.kt [hunk 1]
```diff
@@ -130,6 +135,24 @@ class WebjarsTest {
         }
     }
 
+    @Test
+    fun classifiedAsStatic() {
+        var isStatic = false
+        var location = ""
+        withTestApplication {
```
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/test/io/ktor/server/webjars/WebjarsTest.kt:142]: `withTestApplication { ... }` is deprecated and we try to use new API `testApplication { ... }`

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/http/content/StaticContent.kt [hunk 2]
```diff
@@ -16,6 +16,13 @@ import java.net.*
 import java.nio.file.*
 import kotlin.io.path.*
 
+/**
+ * Attribute to assign the path of a static file served in the response.  The main use of this attribute is to indicate
+ * to subsequent interceptors that a static file was served via the `ApplicationCall.isStaticContent()` extension
+ * function.
+ */
+public val StaticFileLocationProperty: AttributeKey<String> = AttributeKey("StaticFileLocation")
```
Comment (rsinukov) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/server/http/content/StaticContent.kt:24]: do we actually need the path? maybe just add a unit as a marker?

**Review by rsinukov — CHANGES_REQUESTED** [review]
Some minor comments.

Also, whenever you add/change/delete public API, you need to run `./gradlew apiDump` and commit changes to keep track of binary compatibility

File: ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/test/io/ktor/server/webjars/WebjarsTest.kt (hunk 1 above)
Comment (bjhham) [reply ktor-server/ktor-server-plugins/ktor-server-webjars/jvm/test/io/ktor/server/webjars/WebjarsTest.kt:142]: Oh right I meant to ask - in this case where the whole file is the old way would it be better to retrofit to the new method or just leave what's there?

**Review by rsinukov — APPROVED** [review]
LGTM, but please check the codestyle. You can run `./gradlew formatKotlin` locally to auto fix it

**Review by marychatte — APPROVED** [review]
Great work! Also, thanks for test refactoring!

**Pull request merged**